staff/asier/tasks-list_express/src/logic/tasksLogic.js

```diff
@@ -0,0 +1,92 @@
+const tasksData = require('../data/tasksData')
+
+function validateText(text) {
+    if (!text) throw Error('You need to add some text')
+}
+
+function validateUsername(username) {
+    if (!username) throw Error('You need to add an user')
+}
+
+
+const tasksLogic = {
+
+    /* /// create task /// */
+
+    create(text, username) {
+
+        validateText(text)
+        validateUsername(username)
+
+        tasksData.create(text, username)        
+    },
+
+    /* /// List every task /// */
+
+    list() {
+        return tasksData.list()
+
+    },
+
+    /* /// Mark task as done  ///  */
+
+    markDone(id) {       
+        tasksData.markDone(id)
+
+    },
+
+    /* /// Remove task from tasklist */
+
+    remove(id) {
+
+        tasksData.delete(id)         
+    },
+
+    /* /// List tasks already done /// */
+
+    listDone() {
+        return tasksData.listDone()
+    },
+
+    /* /// List tasks pending to do /// */
+
+    listToDo() {
+        return tasksData.listToDo()
+    },
+
+    /* ///  Delete all tasks  /// */
+
+    removeAll() {
+        tasksData.deleteAll()
+    },
+
+    /* /// Update task /// */
+
+    update(id, text, username) {
+
+        if (text || username) {
+
+            tasksData.update(id, text, username)
+
+        } else {
+            validateText(text)
+            validateUsername(username)
+        }
+        
+    },
+
+    /* /// List tasks per user /// */
+
+    userTasks(username) {
+        return tasksData.userTasks(username)
```

Comment:
this is NOT part of the challenge. the idea was just to add a field called username in a task, but not to manage users. managing users is not part of task api.